string path
string audio_source
string user
---
string[] words
string error

